sw/airborne/modules/computer_vision/undistort_image.c

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright (C) 2018, Guido de Croon
+ *
+ * @file modules/computer_vision/undistort_image.c
+ */
+
+// Own header
+#include "modules/computer_vision/undistort_image.h"
+#include <stdio.h>
+#include "modules/computer_vision/lib/vision/image.h"
+#include "modules/computer_vision/lib/vision/undistortion.h"
+
+#ifndef UNDISTORT_FPS
+#define UNDISTORT_FPS 0       ///< Default FPS (zero means run at camera fps)
+#endif
+PRINT_CONFIG_VAR(UNDISTORT_FPS)
+
+#ifndef UNDISTORT_MIN_X_NORMALIZED
+#define UNDISTORT_MIN_X_NORMALIZED -2.0f  ///< Minimal normalized coordinate that will be shown in the undistorted image
+#endif
+PRINT_CONFIG_VAR(UNDISTORT_MIN_X_NORMALIZED)
+
+#ifndef UNDISTORT_MAX_X_NORMALIZED
+#define UNDISTORT_MAX_X_NORMALIZED 2.0f  ///< Maximal normalized coordinate that will be shown in the undistorted image
+#endif
+PRINT_CONFIG_VAR(UNDISTORT_MAX_X_NORMALIZED)
+
+#ifndef UNDISTORT_DHANE_K
+#define UNDISTORT_DHANE_K 1.25f  ///< Maximal normalized coordinate that will be shown in the undistorted image
+#endif
+PRINT_CONFIG_VAR(UNDISTORT_DHANE_K)
+
+#ifndef UNDISTORT_CENTER_RATIO
+#define UNDISTORT_CENTER_RATIO 1.00f  ///< Maximal normalized coordinate that will be shown in the undistorted image
+#endif
+PRINT_CONFIG_VAR(UNDISTORT_CENTER_RATIO)
+
+float min_x_normalized;
+float max_x_normalized;
+float dhane_k;
+float center_ratio;
+
+struct video_listener *listener = NULL;
+
+// Camera calibration matrix (of the Bebop2 for a specific part of the visual sensor):
+static float K[9] = {189.69f, 0.0f, 165.04f,
+                     0.0f, 188.60f, 118.44f,
+                     0.0f, 0.0f, 1.0f};
+
+//static float k;
+
+// Function
+struct image_t *undistort_image_func(struct image_t *img);
```

Comment:
declaration vs definition - probably no need to declare the function, unless you want to put this in a separate header